Correctly check for and use Qt4's qmake

novahahn opened the pull request:
For some reason, qmake is the Qt 5 version on this machine. Compilation failed because it couldn't find the Qt 4 libraries.

Instead of just using qmake, check if either qmake or qmake or qmake-qt4
exists and uses version 4 of Qt. This affects both the dependency check
and the actual compilation of Dwarf Therapist.

This also generally fixes the dependency check. Previously, it actually
tested for Mercurial a second time.

Commit:
Correctly check for and use Qt4's qmake

Instead of just using qmake, check if either qmake or qmake or qmake-qt4
exists and uses version 4 of Qt. This affects both the dependency check
and the actual compilation of Dwarf Therapist.

This also generally fixes the dependency check. Previously, it actually
tested for Mercurial a second time.

Commit:
Query QT_VERSION instead of using qmake -v

andrewd18 commented:
So I pulled my openSuSE 12.3 VM out of the closet and double-checked... sure enough, qmake-qt4 doesn't exist. That's frustrating. I'll pull this in as part of 0.2.4. Thanks for the research.

andrewd18 commented:
Because the "qmake" variable is created as part of the check_dependencies function, the $qmake reference in line 44 will return a blank string if the user runs --skip-deps.

Commit:
Separate qmake detection from dependency check

find_qmake_qt4 is now a separate function called by
build_dwarf_therapist and check_dependencies.

andrewd18 commented:
0.2.4 is live. Thanks for your help.
